src/Workspaces/Core/Portable/Formatting/Rules/NextAction`1.cs

```diff
@@ -7,13 +7,13 @@ namespace Microsoft.CodeAnalysis.Formatting.Rules
     /// <summary>
     /// Represents a next operation to run in a continuation style chaining.
     /// </summary>
-    internal struct NextAction<TArgument>
+    internal readonly struct NextAction<TArgument>
     {
         private readonly int _index;
         private readonly SyntaxNode _node;
-        private readonly IActionHolder<TArgument> _actionCache;
+        private readonly ActionCache<TArgument> _actionCache;
 
-        public NextAction(int index, SyntaxNode node, IActionHolder<TArgument> actionCache)
+        public NextAction(int index, SyntaxNode node, in ActionCache<TArgument> actionCache)
```

Comment:
is this 'in' any benefit, given we're doing the copy below?  or does this prevent two copies from happening?

Reply:
➡️ It does prevent one of the copies.